librad/src/net/protocol/membership/partial_view.rs

```diff
@@ -127,12 +127,16 @@ where
             return vec![];
         }
 
-        let demoted = if self.num_active() >= self.max_active {
+        let demoted = if self.is_active_full() {
             self.demote_random()
         } else {
             vec![]
         };
 
+        if self.is_passive(&info.peer_id) {
+            self.passive.remove(&info.peer_id);
```

Comment:
This needs to add an `Evicted` transition to the return value, for example by calling `evict`.